modules/base/acl/module.py

```diff
@@ -25,494 +23,606 @@ def __init__(self, bot):
+            header={
+                "role": _(ctx, "Role"),
+                "level": _(ctx, "Level"),
+            },
+        )
 
-        To unlink the group from the parent, set it to "".
+        for page in table:
+            await ctx.send("```" + page + "```")
 
-        To set up virtual group with no link to Discord roles, set ``role_id``
-        to ``0``.
-        """
-        RE_NAME = r"[a-zA-Z-]+"
-        if re.fullmatch(RE_NAME, name) is None:
+    @check.acl2(check.ACLevel.GUILD_OWNER)
+    @acl_mapping_.command(name="add")
+    async def acl_mapping_add(self, ctx, role: nextcord.Role, level: str):
+        """Add ACL level to role mappings."""
+        try:
+            level: ACLevel = ACLevel[level]
+        except KeyError:
             await ctx.reply(
-                _(ctx, "Bad group name (regex `{regex}`)").format(regex=RE_NAME)
+                _(ctx, "Invalid level. Possible options are: {keys}.").format(
+                    keys=", ".join(f"'{key}'" for key in ACLevel.__members__.keys())
+                )
             )
             return
 
-        if len(parent) == 0:
-            parent = None
+        m = ACLevelMappping.add(ctx.guild.id, role.id, level)
+        if m is None:
+            await ctx.reply(_(ctx, "That role is already mapped to some level."))
+            return
 
-        group = ACL_group.add(ctx.guild.id, name, parent, role_id)
-        await ctx.reply(embed=self.get_group_embed(ctx, group))
-        await guild_log.warning(
+        await ctx.reply(
+            _(ctx, "Role **{role}** will be mapped to **{level}**.").format(
+                role=role.name, level=level.name
+            )
+        )
+        await guild_log.info(
             ctx.author,
             ctx.channel,
-            f"New ACL group '{name}' (parent '{parent}', role ID '{role_id}').",
+            f"New ACLevel mapping from role '{role.name}' to level '{level.name}'.",
         )
 
-    @commands.check(check.acl)
-    @acl_group.command(name="update")
-    async def acl_group_update(self, ctx, name: str, param: str, value):
-        """Update ACL group.
+    @check.acl2(check.ACLevel.GUILD_OWNER)
+    @acl_mapping_.command(name="remove")
+    async def acl_mapping_remove(self, ctx, role: nextcord.Role):
+        """Remove ACL level to role mapping."""
+        removed: bool = ACLevelMappping.remove(ctx.guild.id, role.id)
+        if not removed:
+            await ctx.reply(_(ctx, "That role is not mapped to any level."))
+            return
 
-        :param name: name of group
+        await ctx.reply(_(ctx, "Role mapping was sucessfully removed."))
+        await guild_log.info(
+            ctx.author,
+            ctx.channel,
+            f"ACLevel mapping for role '{role.name}' removed.",
+        )
 
-        ``param`` -- ``value`` options:
+    @check.acl2(check.ACLevel.SUBMOD)
+    @acl_.group(name="default")
+    async def acl_default_(self, ctx):
+        """Manage ACLevel defaults of commands."""
```

Comment:
What are defaults good for? I'm, as a person, who sees this first, kinda confused, without looking for docs.

Reply:
I'm still not sure about the command names. This is used to overwrite hardcoded defaults:
```py
@check.acl2(check.ACLevel.MEMBER)
async def foo(self, ctx):
    ...
```
can be updated by calling `!acl default set foo EVERYONE`. Then you don't need to have any role to use that command, if that's what you want/need on your server.

ACL2 provides sane defaults, but guild owners should have the option to change the defaults.

Reply:
Oh, I see. In this case I think that this should be said in short version in this group description. Something like:

```Python
"""Manage ACL default level assignment for commands."""```

Something like that?